sparse/server.go

```diff
@@ -1,18 +1,83 @@
 package sparse
 
 import (
-	"crypto/sha1"
-	"encoding/binary"
+	"bytes"
 	"encoding/gob"
-	"hash"
+	"fmt"
 	"net"
 	"os"
+	"sort"
 	"strconv"
 	"time"
 
-	"github.com/rancher/sparse-tools/log"
+	log "github.com/Sirupsen/logrus"
 )
 
+const verboseServer = true
+
+type requestCode int
+
+const (
+	requestMagic    requestCode = 31415926
+	syncRequestCode requestCode = 1 + iota
+	syncHole
+	syncData
+	syncDone
+)
+
+func (reqCode requestCode) String() string {
+	var s string
+	if reqCode&syncRequestCode == syncRequestCode {
```

Comment:
TOTALLY WRONG!

The variable you assigned is incremental, but the way you're judging is bitwise? 

syncRequestCode = 2(`10b`), syncHole =3(`11b`), syncData = 4(`100b`), syncDone = 5(`101b`)

If `reqCode = syncHole = 3`, then `reqCode & syncRequestCode = 11b & 10b = 10b = syncRequestCode`
If `reqCode = syncDone = 5`, then `reqCode & syncData = 101b & 100b = 100b = syncData`

How can this possibly work???

I have no idea why you want to use bitwise operation here, it makes no sense